### PR TITLE
Remove blank lines instead of emptying it's contents

### LINKED ISSUE #664: Remove emptied lines instead of emptying only the contents of the line
image (1)

Polyglot piranha currently empties a line instead of removing this line. This means we will have to manually delete all such blank lines post polyglot execution. 

Running linters might not be straightforward on larger repositories as it might cleanup unmodified lines as well leading to a huge diff.

### PR DESCRIPTION
**Pull request by Bennet-Sunder**
Fixes #664

### CODE REVIEW

File: src/models/matches.rs [hunk 1]
```diff
@@ -51,6 +51,10 @@ pub struct Match {
   #[get_mut]
   #[serde(skip)]
   pub(crate) associated_comments: Vec<Range>,
+  #[get]
+  #[get_mut]
+  #[serde(skip)]
```
Comment (ketkarameya) [inline src/models/matches.rs:56]: this is not just empty lines right? All leading white spaes?

**Comment by Bennet-Sunder** [comment]
@dvmarcilio Could you pls have a look at this PR when you have bandwidth.

File: src/models/unit_tests/source_code_unit_test.rs [hunk 2]
```diff
@@ -699,3 +699,77 @@ fn test_satisfies_outermost_enclosing_node() {
...
+      .first()
+      .unwrap(),
+    Range {
+      start_byte: 15,
+      end_byte: 24,
+      start_point: Point { row: 0, column: 15 },
+      end_point: Point { row: 1, column: 8 }
+    }
+  );
+}
```
Comment (danieltrt) [inline src/models/unit_tests/source_code_unit_test.rs:775]: Hi @Bennet-Sunder ,

Thank you for your contribution! I tested this internally and noticed that the change might break some rewrites. The issue is that in some grammars (particularly Swift), whitespace and newlines are part of the node. If you delete them, you might accidentally delete parts of a sibling node unrelated to the node you're deleting.

Please accept this test suggestion to see what I mean. The test should fail with your change (make sure to add Swift at the top). My proposed solution is to delete only whitespace or newlines until you find the end_byte of the previous sibling node. Let me know if you need help with this.

```suggestion
}

#[test]
fn test_switch_entry_blank_lines() {
  let inline_cleanup_rule = piranha_rule! {
    name= "inline_cleanup_rule",
    query= "
    (
        (switch_entry
            (switch_pattern) @p1
            (switch_pattern) @p2
            (switch_pattern) @p3
            (switch_pattern) @p4
            (switch_pattern) @p5
            (switch_pattern) @p6
        ) @custom_entry
        (#eq? @p1 \".case_zeta_first\")
    )
    ",
    replace_node = "custom_entry",
    replace = ""
  };

  let inline_rule = InstantiatedRule::new(&inline_cleanup_rule, &HashMap::new());

  let source_code = r#"
    public var namespace: ParameterNamespace {
        switch self {
        case .case_random_word_alpha,
             .case_random_word_beta,
             .case_random_word_gamma,
             .case_random_word_delta:
            return .namespace_group_alpha
        case .case_zeta_first,
             .case_zeta_second,
             .case_zeta_third,
             .case_zeta_fourth,
             .case_zeta_fifth,
             .case_zeta_sixth:
            return .namespace_group_beta
        case .case_random_word_omega:
            return .namespace_group_gamma
        }
    }
  "#
  .trim();

  let piranha_arguments = PiranhaArgumentsBuilder::default()
    .paths_to_codebase(vec![UNUSED_CODE_PATH.to_string()])
    .language(PiranhaLanguage::from(SWIFT))
    .build();

  let mut rule_store = RuleStore::new(&piranha_arguments);
  let mut parser = piranha_arguments.language().parser();

  let mut source_code_unit = SourceCodeUnit::new(
    &mut parser,
    source_code.to_string(),
    &HashMap::new(),
    PathBuf::new().as_path(),
    &piranha_arguments,
  );

  source_code_unit.apply_rule(inline_rule, &mut rule_store, &mut parser, &None);
  let transformed_code = source_code_unit.code();

  let expected_code = r#"
    public var namespace: ParameterNamespace {
        switch self {
        case .case_random_word_alpha,
             .case_random_word_beta,
             .case_random_word_gamma,
             .case_random_word_delta:
            return .namespace_group_alpha
        case .case_random_word_omega:
            return .namespace_group_gamma
        }
    }
  "#
  .trim();
  assert_eq!(transformed_code, expected_code);
}

```

File: src/models/unit_tests/source_code_unit_test.rs [hunk 3]
```diff
@@ -16,7 +16,7 @@ use tree_sitter::Parser;
 use crate::{
   filter,
   models::{
-    default_configs::{JAVA, UNUSED_CODE_PATH},
+    default_configs::{JAVA, RUBY, UNUSED_CODE_PATH},
```
Comment (danieltrt) [inline src/models/unit_tests/source_code_unit_test.rs:19]:
```suggestion
    default_configs::{JAVA, RUBY, SWIFT, UNUSED_CODE_PATH},
```

**Review by danieltrt — CHANGES_REQUESTED** [review]

**Comment by Bennet-Sunder** [comment]
@stefanheule @danieltrt Thanks for your review of the PR. I can see that a new PR is raised with the fixes. I'll close this PR in favour of the newer one.

**Pull request closed without merging**